crates/katana/core/Cargo.toml

```diff
@@ -37,11 +38,22 @@ tokio.workspace = true
 tracing.workspace = true
 url.workspace = true
 
+alloy-primitives = { version = "0.6.4", default-features = false }
+alloy-sol-types = { version = "0.6.4", default-features = false, features = ["json"] }
+
+alloy-transport = { version = "0.1.0", default-features = false }
+alloy-provider = { version = "0.1.0", default-features = false }
+alloy-signer = { version = "0.1.0", default-features = false }
+alloy-network = { version = "0.1.0", default-features = false }
+alloy-rpc-types = { version = "0.1.0", default-features = false }
+alloy-signer-wallet = { version = "0.1.0", default-features = false }
+alloy-contract = { version = "0.1.0", default-features = false }
```

Comment:
except for the primitives, can we make the others optional ? considering its only used in `messaging` feature

Reply:
Done